src/main/java/org/lanternpowered/server/event/LanternEventManager.java

```diff
@@ -274,11 +274,9 @@ private void unregister(Predicate<RegisteredListener<?>> unregister) {
             while (it.hasNext()) {
                 final RegisteredListener<?> listener = it.next();
                 if (unregister.test(listener)) {
-                    if (listener.getHandle() instanceof AnnotatedEventListener) {
                         synchronized (this.registeredListeners) {
-                            this.registeredListeners.remove(((AnnotatedEventListener) listener.getHandle()).getHandle());
+                            this.registeredListeners.remove(listener.getHandle());
                         }
```

Comment:
Can you fix the formatting of the synchronized block, it's 4 spaces to much to the right. After that, I can merge.